webpack/components/AnsibleRolesSwitcher/AnsibleRolesSwitcherReducer.js

```diff
@@ -0,0 +1,91 @@
+import Immutable from 'seamless-immutable';
+
+import {
+  ANSIBLE_ROLES_REQUEST,
+  ANSIBLE_ROLES_SUCCESS,
+  ANSIBLE_ROLES_FAILURE,
+  ANSIBLE_ROLES_ADD,
+  ANSIBLE_ROLES_REMOVE,
+  ANSIBLE_ROLES_ASSIGNED_PAGE_CHANGE,
+} from './AnsibleRolesSwitcherConstants';
+
+const ansibleRolesSuccess = (state, payload) => {
+  const {
+    page,
+    perPage,
+    subtotal,
+    results,
+    initialAssignedRoles,
+    inheritedRoleIds,
+  } = payload;
+
+  return state.merge({
+    loading: false,
+    itemCount: Number(subtotal),
+    pagination: { page: Number(page), perPage: Number(perPage) },
+    results,
+    assignedRoles: initialAssignedRoles,
+    inheritedRoleIds,
+  });
+};
+
+const addItem = (list, item) => ([...(list || []), item]);
+
+const removeItem = (list, item) => list.filter(listItem => item.id !== listItem.id);
+
+const removeItemNewState = (state, role) => ({
+  assignedRoles: removeItem(state.assignedRoles, role),
+  results: addItem(state.results, role),
+  itemCount: state.itemCount + 1,
+});
+
+const addItemNewState = (state, role) => ({
+  assignedRoles: addItem(state.assignedRoles, role),
+  itemCount: state.itemCount - 1,
```

Comment:
Am I missing something? shouldn't it do `itemCount + 1` while the remove should do the `itemCount - 1`?

Reply:
`itemCount` tracks the item count of the unassigned roles for pagination to display correct values, so if we add a role to a host, we need to decrement the unassigned roles by 1.